fix phpstan

thibdrev opened the pull request:
Function dol_get_first_day() should return int but returns string. 
Function getWeekNumber() should return int but returns string.

thibdrev committed:
phpstan

Function dol_get_first_day() should return int but returns string.
Function getWeekNumber() should return int but returns string.

eldy commented on htdocs/core/lib/date.lib.php:
```diff
@@ -579,7 +579,7 @@ function dol_get_first_day($year, $month = 1, $gm = false)
 		return '';
 	}
-	return dol_mktime(0, 0, 0, $month, 1, $year, $gm);
+	return (int) dol_mktime(0, 0, 0, $month, 1, $year, $gm);
```
dol_mktime can return
int|string					Date as a timestamp, '' if error

We can modify this to return
int|null					Date as a timestamp, null if error

but the non int case must remain propagated.

thibdrev replied:
Right, I missed that sorry.
I've changed it to `@return int|string` since we have a `return '';` at line 580 which I missed as well.

thibdrev committed:
phpstan

Function dol_get_first_day() should return int but returns string.
Function dol_get_last_day() should return int but returns string.
